### PR TITLE
Object Tags APIs - Set, Get, Remove.

### PR DESCRIPTION
**Pull request by BigUstad**
Example, Functional test.

### CODE REVIEW

File: Minio/DataModel/Tagging.cs [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+            String str = String.Empty;
+
+            try
+            {
+                settings = new XmlWriterSettings();
+                settings.OmitXmlDeclaration = true;
+
+                ns = new XmlSerializerNamespaces();
+                // ns.Add("", "http://s3.amazonaws.com/doc/2006-03-01/");
```
Comment (kannappanr) [inline Minio/DataModel/Tagging.cs:128]:
```suggestion
```

File: Minio/DataModel/Tagging.cs [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+                ns.Add(string.Empty, string.Empty);
+
+                StringWriter sw = new StringWriter(CultureInfo.InvariantCulture);
+
+                xs = new XmlSerializer(typeof(Tagging), "");
+                xw = XmlWriter.Create(sw, settings);
+                xs.Serialize(xw, this, ns);
+                //xs.Serialize(xw, this);
```
Comment (kannappanr) [inline Minio/DataModel/Tagging.cs:136]:
```suggestion
```

File: Minio/DataModel/Tagging.cs [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+                //xs.Serialize(xw, this, ns);
```
Comment (kannappanr) [inline Minio/DataModel/Tagging.cs:137]:
```suggestion
```

**Review by sinhaashish — CHANGES_REQUESTED** [review]
please check the new line in all the edited files

**Comment by BigUstad** [comment]
> please check the new line in all the edited files

The newline at the end of the file is not enforced as per C# compiler rules. I've tried to not have newline at the end of the file in minio-dotnet as I observed it in files before.

File: Minio/DataModel/Tagging.cs (hunk 1 above)
Comment (BigUstad) [reply Minio/DataModel/Tagging.cs]: Removed.

File: Minio.Examples/Cases/GetObjectTags.cs [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+                Console.WriteLine("Running example for API: GetObjectTags");
+                var tags = await minio.GetObjectTagsAsync(
+                    new GetObjectTagsArgs()
+                        .WithBucket(bucketName)
+                        .WithObject(objectName)
+                        .WithVersionId(versionId)
+                );
+                if (tags != null && tags.GetTags() != null && tags.GetTags().Count > 0)
+                {
+                    Console.WriteLine($"Got Object Tags set for bucket {bucketName}/{objectName}.");
```
[the remark this answers is not in the document]
Comment (BigUstad) [reply Minio.Examples/Cases/GetObjectTags.cs]: Change made..PTAL.

File: Minio.Examples/Cases/GetObjectTags.cs [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+                Console.WriteLine($"Object Tags not set for {bucketName}/{objectName}.");
+                Console.WriteLine();
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine($"[Object]  Exception: {e}");
+            }
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (BigUstad) [reply Minio.Examples/Cases/GetObjectTags.cs:57]: Plz refer https://github.com/minio/minio-dotnet/pull/503#issuecomment-744943647

File: Minio.Examples/Cases/RemoveBucketTags.cs [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+ * limitations under the License.
+ */
+using System;
+using System.Threading.Tasks;
+
+namespace Minio.Examples.Cases
+{
+    public class RemoveBucketTags
+    {
+        // Put Encryption Configuration for the bucket
```
[the remark this answers is not in the document]
Comment (BigUstad) [reply Minio.Examples/Cases/RemoveBucketTags.cs]: Change made. PTAL.

File: Minio.Examples/Cases/RemoveObjectTags.cs [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+ */
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Minio.Examples.Cases
+{
+    public class RemoveObjectTags
+    {
+        // Put Encryption Configuration for the bucket
```
[the remark this answers is not in the document]
Comment (BigUstad) [reply Minio.Examples/Cases/RemoveObjectTags.cs]: Changed.

File: Minio.Examples/Cases/SetObjectTags.cs [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+ */
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Minio.Examples.Cases
+{
+    public class SetObjectTags
+    {
+        // Put Encryption Configuration for the bucket
```
[the remark this answers is not in the document]
Comment (BigUstad) [reply Minio.Examples/Cases/SetObjectTags.cs]: Changed/Updated.

**Review by sinhaashish — APPROVED** [review]

File: Minio.Examples/Cases/RemoveObjectTags.cs [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+ * limitations under the License.
+ */
+using System;
+using System.Threading.Tasks;
+
+namespace Minio.Examples.Cases
+{
+    public class RemoveObjectTags
+    {
+        // Remove Tags set for the bucket
```
Comment (kannappanr) [inline Minio.Examples/Cases/RemoveObjectTags.cs:23]:
```suggestion
```

Comment (BigUstad) [reply Minio.Examples/Cases/RemoveObjectTags.cs]: Comment updated

**Review by kannappanr — APPROVED** [review]
LGTM

**Pull request merged**